web/src/utils/analytics.ts

```diff
@@ -38,7 +38,33 @@ export enum ShareType {
 export const trackShare = (shareType: ShareType) => () =>
   trackEvent(TrackEvent.SHARE_BUTTON_CLICKED, { shareType });
 
-export const trackShareChart = (shareType: ShareType) => () =>
+export const trackShareChart = (shareType: ShareType, chartId: Charts) => () =>
   trackEvent(TrackEvent.SHARE_CHART, {
     shareType,
+    chartId,
   });
+
+interface TrackChartSharesByShareType {
+  [chartId: string]: {
+    [shareType: string]: () => void;
+  };
+}
+
+const makeTrackChartShareFunctions = () => {
+  const trackChartShareByType: TrackChartSharesByShareType = {};
+
+  for (const chartId of Object.values(Charts)) {
+    for (const shareType of Object.values(ShareType)) {
+      if (!(chartId in trackChartShareByType)) {
+        trackChartShareByType[chartId] = {};
+      }
+      trackChartShareByType[chartId][shareType] = trackShareChart(shareType, chartId);
+    }
+  }
+
+  return trackChartShareByType;
+};
```

Comment:
Wanna use fancy object functions? 

```suggestion
const makeTrackChartShareFunctions = () =>
  Object.fromEntries(
    Object.values(Charts).map((chartId) => [
      chartId,
      Object.fromEntries(
        Object.values(ShareType).map((shareType) => [
          shareType,
          trackShareChart(shareType, chartId),
        ])
      ),
    ])
  );
```

Should be identical functionality wise but please double check!

Reply:
>Wanna use fancy object functions?

Why yes, yes I do :D